### PR TITLE
Ignore event spams of keys we already have

### PR DESCRIPTION
**Pull request by neithernut**
This PR fixes a bug where keeping a key down for a while causes the DAG not to be traverses correctly.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Will test, stay tuned!

File: src/input/hotkeys.c [hunk 1]
```diff
@@ -118,9 +118,27 @@ ws_hotkeys_eval(
...
+            if (buf->code != code) {
+                // event not of interest
+                continue;
+            }
+            if (buf->value != ev->value) {
+                // the button was release before
+                break;
+            }
+            // we consume the event but we'll _not_ store it or iterate the DAG
+            return empty_eventlist();
```
Comment (matthiasbeyer) [inline src/input/hotkeys.c:138]: Wouldn't it result in less code if you'd do a `if (buf->code == code) { return ... }` here and remove the `continue` therefor?

Comment (neithernut) [inline src/input/hotkeys.c:138]: The extra level of intendation would make the documentation in line 137 a pain in the ass.

Comment (matthiasbeyer) [inline src/input/hotkeys.c:138]: Meh. Could be fixed using a multi-line arg, but well.. I won't argue about that.

**Comment by matthiasbeyer** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
**STOP!**

Not ACK. Press a key twice crashes the system.

**Comment by neithernut** [comment]
May this be related to #514? (Will incluse a fix in this PR btw.)

**Comment by neithernut** [comment]
Please check whether there are still problems.

**Comment by TheNeikos** [comment]
I can't reproduce it

**Comment by TheNeikos** [comment]
#515 is still open, do you think that is related?

**Comment by neithernut** [comment]
Issue #515 _should_ now be resolved (I only tested it briefly).

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**